tests/server/query_params/test_filter.py

```diff
@@ -383,6 +383,31 @@ def test_not_or_and_precedence(check_response):
     check_response(request, expected_ids)
 
 
```

Comment:
Would it be possible to translate a couple of the more complicated mongo-specific tests you've added into OPTIMADE queries we can run against our test data (so we can compare what is returned by Mongo vs Elastic)? I think this is the most useful way of testing the new functionality.

Reply:
I have added some more test. Apart from the Elasticsearch implementation not supporting queries on references, all tests pass for Elasticsearch.